.github/workflows/integ-test.yml

```diff
@@ -65,6 +67,13 @@ jobs:
           ls -al tests/integration/integration_config.yml
       - run: cd $WORKDIR && ansible-playbook tests/integration/prepare/prepare_iso.yml
       - run: cd $WORKDIR && ansible-playbook tests/integration/prepare/prepare_vm.yml
+
+      - id: set-matrix
+        shell: bash
+        run: |-
+          echo "matrix=$(
+            ls -r $WORKDIR/examples | jq -R -s -c 'split("\n")[:-1]'
+          )" >> $GITHUB_OUTPUT
```

Comment:
This will be run only after `prepare_iso.yml`, `prepare_vm.yml`. Then, relatively fast test-examples will need to wait on those slow preparations. Move this to separate job. One job just to generate matrix, another job just to run `ansible-playbook --check examples/$example_name`.

Reply:
To generate the matrix, I made a separate job ``examples-matrix`` as suggested. Then I also create a separate job to run those examples - ``examples-run``.